examples/example2.py

```diff
@@ -11,13 +11,12 @@
 
     v1 = client.CoreV1Api()
     count = 10
-    w = watch.Watch()
-
-    async for event in w.stream(v1.list_namespace, timeout_seconds=10):
-        print("Event: {} {}".format(event['type'], event['object'].metadata.name))
-        count -= 1
-        if not count:
-            w.stop()
+    async with watch.Watch() as w:
```

Comment:
Please leave examples (example2, example4) unchanged because these show that the library is almost identical with synchronous version (https://github.com/kubernetes-client/python/blob/master/examples/example2.py). You can add additional examples with the context manager if you want.

Reply:
It is not so identical... The synchronous version uses a generator with `finally` clause, and the generator will be closed by gc after their usage. Async generators are generally not collected automatically because they are referenced by EventLoop, and `Watch` is not an async generator, it is only an async iterator, so an explicit close is needed.

We may keep both lines there to inform the users that they can use the identical way, but context manager is recommended to prevent resource leaking.

Reply:
Good point, so definitely the asynchronous generator is the best solution, but it's been introduced in Python 3.6. For the time being we can leave it as is.

Reply:
@tomplus so what about the examples?

Reply:
I suggest removing context manager from example2. Optionally we can add `w.close()` explicit with a comment that context-manager can handle it better (see example4). Example4 shows usage of context-manager as you suggested. Thanks.

Reply:
OK